File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/ServiceProvider/commerce-delivery-catalog/v1.0/Product.js
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import AJAX from '../../../utilities/AJAX/index';

const VERSION = 'v1.0';

function resolveProductsPath(basePath, channelId) {
	return `${basePath}${VERSION}/channels/${channelId}/products`;
}

export default function Product(basePath) {
	return {
		getBaseURL: (channelId) => resolveProductsPath(basePath, channelId),
		getProductsByChannelId: (channelId, ...params) =>
			AJAX.GET(resolveProductsPath(basePath, channelId), ...params),
	};
}

File: modules/apps/commerce/commerce-order-rule-web/src/main/java/com/liferay/commerce/order/rule/web/internal/frontend/model/Product.java
/**
 * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.commerce.order.rule.web.internal.frontend.model;

/**
 * @author Alessio Antonio Rendina
 */
public class Product {

	public Product(long cProductId, String name) {
		_cProductId = cProductId;
		_name = name;
	}

	public long getCProductId() {
		return _cProductId;
	}

	public String getName() {
		return _name;
	}

	private final long _cProductId;
	private final String _name;

}